### PR TITLE
Feat/kadena cli/txlocal

### PR DESCRIPTION
**Pull request by barthuijgen**
- add tx local command that can be used to quickly test pact code

### CODE REVIEW

File: packages/tools/kadena-cli/src/tx/commands/templates/templates.ts [hunk 1]
```diff
@@ -50,6 +50,24 @@ networkId: "{{network:networkId}}"
...
+  code: string,
+  chainId: string,
+  network: string,
+): string => `
+code: |-
+  ${code}
+data:
+meta:
+  chainId: "${chainId}"
+  gasLimit: 2300
```
Comment (nillo) [inline packages/tools/kadena-cli/src/tx/commands/templates/templates.ts:63]: I would make the gaslimit optional also. When sending large bits of code you might have to increase it.

File: packages/tools/kadena-cli/src/tx/commands/txLocal.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    async (option, { values }) => {
+      const { network, networkConfig } = await option.network();
+      const { chainId } = await option.chainId();
+      log.debug('tx:local', { network, networkConfig, chainId, values });
+
+      const code = values[0];
+      const templateChainId = (chainId ?? '0') as ChainId;
+      const templateNetworkId = networkConfig?.networkId ?? 'testnet04';
+      const templateNetworkHost =
+        networkConfig?.networkHost ?? 'https://api.testnet.chainweb.com';
```
Comment (nillo) [inline packages/tools/kadena-cli/src/tx/commands/txLocal.ts:30]: Why is this here, its been a while, but if I recall correctly when none is give it automatically would fall back to this. Well at least it used to.

File: packages/tools/kadena-cli/src/tx/commands/txLocal.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          chainId: templateChainId,
+          network: templateNetwork,
+          networkId: templateNetworkId,
+          networkHost: templateNetworkHost,
+          networkExplorerUrl: '',
+        }),
+      );
+
+      const response = await client
+        .local(transaction, {
```
Comment (nillo) [inline packages/tools/kadena-cli/src/tx/commands/txLocal.ts:54]: I forgot the function but kadena client exposes another function, that has this automatically disabled and does the call , saves u padding 2 options.

File: packages/tools/kadena-cli/src/utils/globalOptions.ts [hunk 4]
```diff
@@ -99,6 +99,29 @@ export const globalOptions = {
...
+    ),
+    expand: async (network: string) => {
+      if (network === undefined) return null;
+      try {
+        return await loadNetworkConfig(network);
+      } catch (e) {
+        log.info(
+          `\nNo configuration for network "${network}" found. Please configure the network.\n`,
+        );
+        await program.parseAsync(['', '', 'networks', 'create']);
```
Comment (nillo) [inline packages/tools/kadena-cli/src/utils/globalOptions.ts:120]: This should not be in here anymore

File: packages/tools/kadena-cli/src/tx/commands/txLocal.ts (hunk 3 above)
Comment (ash-vd) [reply packages/tools/kadena-cli/src/tx/commands/txLocal.ts:54]: That's dirtyRead: https://github.com/kadena-community/kadena.js/blob/main/packages/libs/client/src/client/client.ts#L392

File: packages/tools/kadena-cli/src/tx/commands/txLocal.ts (hunk 2 above)
Comment (barthuijgen) [reply packages/tools/kadena-cli/src/tx/commands/txLocal.ts:30]: Yeah I wasn't sure how to do this. The problem is I want the network flag to be optional and not prompted. So unless you pass `-n mainnet` as example, you won't have an `networkConfig` object. 

I can change it so it fetches one of the existing networks instead of hardcoding default values here. But then we need to discuss the logic of picking a network

File: packages/tools/kadena-cli/src/tx/commands/txLocal.ts (hunk 3 above)
Comment (barthuijgen) [reply packages/tools/kadena-cli/src/tx/commands/txLocal.ts:54]: Personally I like local better, having the option there makes it clear I'm doing a local call with those settings. dirtyRead doesn't mean anything to me. But if it's against Kadena's best practices I can change it...

File: packages/tools/kadena-cli/src/tx/commands/templates/templates.ts (hunk 1 above)
Comment (barthuijgen) [reply packages/tools/kadena-cli/src/tx/commands/templates/templates.ts:63]: is it even possible to submit a transaction without gasLimit? shouldn't I just hardcode it at a very high value then?

File: packages/tools/kadena-cli/src/tx/commands/txLocal.ts (hunk 3 above)
Comment (nillo) [reply packages/tools/kadena-cli/src/tx/commands/txLocal.ts:54]: We can go with local dirtyRead in naming doesn’t say anything idd we should actually rename that function to something more obvious in the client. Let’s keep local for now for us

File: packages/tools/kadena-cli/src/utils/globalOptions.ts [hunk 5]
```diff
@@ -139,6 +162,27 @@ export const globalOptions = {
       }
     },
   }),
+  chainIdOptional: createOption({
```
Comment (realdreamer) [inline packages/tools/kadena-cli/src/utils/globalOptions.ts:165]: instead of creating this option completely - we can do like this right `globalOptions.chainId({ isOptional: true })`  and then adding a change in transform what you've added now `if (!chainId) return null;` it works, do we really need separate optional one?

File: packages/tools/kadena-cli/src/tx/commands/templates/templates.ts (hunk 1 above)
Comment (nillo) [reply packages/tools/kadena-cli/src/tx/commands/templates/templates.ts]: optional wasn't the right word, should have been configurable. we can keep this as a default but to option to increase the value should be there

File: packages/tools/kadena-cli/src/tx/commands/txLocal.ts (hunk 2 above)
Comment (nillo) [reply packages/tools/kadena-cli/src/tx/commands/txLocal.ts]: we can go for this. it should fall back to something, wtihout going through all those hassles

File: packages/tools/kadena-cli/src/utils/globalOptions.ts (hunk 5 above)
Comment (barthuijgen) [reply packages/tools/kadena-cli/src/utils/globalOptions.ts:160]: Yes because when you add `if (!chainId) return null;` it also affects the return type. I will need to go and add guards and errors everywhere that network is used.

**Review by nillo — APPROVED** [review]
